# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a=101
b=210
c=111
d=402
(a+b)*c/d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [6]:
a=11283
b=89

c=a % b
print(c)


69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [7]:
d=11283//89
print (d)

126


4. Запишіть у змінну `name` ваше імʼя.

In [9]:
name="Natalia"
print(name)

Natalia


5. Запишіть у змінну `surname` ваше прізвище.

In [11]:
surname="Bitiutska"
print(surname)

Bitiutska


6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [12]:
name_surname=name+" "+surname
print(name_surname)

Natalia Bitiutska


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [14]:
print(name_surname.upper())

NATALIA BITIUTSKA


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

In [20]:
name_surname[2]

't'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [32]:
name_surname=name+" "+surname
print(name_surname)
print(name_surname[2])

Natalia Bitiutska
t


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [33]:
print(name_surname[0])
print(name_surname[-1])

N
a


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [17]:
print(len(surname))

9


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [28]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
print(my_string.startswith("The"))
print(my_string.endswith("dog!"))


True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [31]:
my_string = "The quick brown fox jumps over the lazy dog!!!"
s1="The"
s2="quick"
s3="brown"
s4="fox"
s5="jumps"
s6="over"
s7="the"
s8="lazy"
s9="dog!!!"
s10=s1+"_"+s2+"_"+s3+"_"+s4+"_"+s5+"_"+s6+"_"+s7+"_"+s8+"_"+s9
print(s10)

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`